Changelog.md

```diff
@@ -4,6 +4,8 @@ The change log describes what been "Added", "Removed", "Changed" or "Fixed" betw
 
 ### Unreleased
 
+- The `setHttpClient`, `setServerToken` and `getServerToken` methods have been removed from `Stampie\MailerInterface`.
+- The `Stampie\MailerInterface::send` method no longer return a boolean. Errors are now reported using exceptions that implements `Stampie\ExceptionInterface`.
```

Comment:
I would add a note explaining that core implementations were already throwing exception here (to explain that our BC break here is minimal)